Core/include/Acts/Navigation/TryAllNavigationPolicies.hpp

```diff
@@ -0,0 +1,72 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Navigation/INavigationPolicy.hpp"
+
+namespace Acts {
+
+class TrackingVolume;
+class GeometryContext;
+class Logger;
+
+/// Policy which adds **all** portals of the volume to the main navigation
+/// stream
+class TryAllPortalNavigationPolicy final : public INavigationPolicy {
+ public:
```

Comment:
couldn't this be merged with `TryAllSurfaceNavigationPolicy` and configurable if we insert only portals or only sensitives?

Reply:
Could to, yes